### PR TITLE
[ENH] Add Schaefer atlas

### LINKED ISSUE #1667: Add Schaefer/Yeo atlas
There is rapidly increasing adoption of the SchaeferYeo atlas now in the imaging neuroscience community. It would be great to add it soon in nilearn.datasets has been precomputed with 100, 200, 300, ... 1000 parcellations including their assignment to the original 7 and 17 Yeo networks.

https://github.com/ThomasYeoLab/CBIG

### PR DESCRIPTION
**Pull request by fliem**
Closes #1667 

What do you think?

Let me know if the next release is not 0.5.1, in that case the `versionadded` needs to be changed.

### CODE REVIEW

**Comment by fliem** [comment]
I'm also adding the atlas creators @alexandschaefer @ahpxue

File: nilearn/datasets/atlas.py [hunk 1]
```diff
@@ -1185,3 +1186,107 @@ def fetch_atlas_pauli_2017(version='prob', data_dir=None, verbose=1):
     return Bunch(maps=atlas_file,
                  labels=labels,
                  description=fdescr)
+
+
+
+def fetch_atlas_schaefer_2018(n_rois=100, yeo_networks=7, resolution_mm=1,
```
Comment (alexandschaefer) [inline nilearn/datasets/atlas.py]: I would somehow prefer the 400 as default. As this was our main parcellation in the paper.

File: nilearn/datasets/atlas.py [hunk 2]
```diff
@@ -1185,3 +1186,107 @@ def fetch_atlas_pauli_2017(version='prob', data_dir=None, verbose=1):
...
+                    'Parcellations/MNI/'
+                    )
+
+    opts = {}
+    files = []
+    labels_file_template = 'Schaefer2018_{}Parcels_{}Networks_order.txt'
+    img_file_template = 'Schaefer2018_{}Parcels_{}Networks_order_FSLMNI152_{}mm.nii.gz'
+    for f in [labels_file_template.format(n_rois, yeo_networks),
+              img_file_template.format(n_rois, yeo_networks, resolution_mm)]:
+        files.append((f, base_url + f, opts))
```
Comment (alexandschaefer) [inline nilearn/datasets/atlas.py]: What is opts used for? I dont get it atm.

**Comment by fliem** [comment]
Didn't realize that pandas is not a dependency. Changed the code to pd.read_csv to np.genfromtxt

File: nilearn/datasets/atlas.py (hunk 1 above)
Comment (fliem) [reply nilearn/datasets/atlas.py]: Thanks Alex. Changed to 400

File: nilearn/datasets/atlas.py (hunk 2 above)
Comment (fliem) [reply nilearn/datasets/atlas.py]: cleaned that up

**Comment by kchawla-pi** [comment]
Inform users about the new addition in the `whats_new.rst`

**Comment by fliem** [comment]
> Inform users about the new addition in the `whats_new.rst`

Done

**Comment by fliem** [comment]
hi @kchawla-pi 
on the last push one appveyor job failed (probably because of neurovaul not responding). I re-triggered the CI, now all checks have passed. So this is ready as well.

**Comment by kchawla-pi** [comment]
Thanks for your work @fliem !
Ready for a final review & check. I'll give people a day to respond and then merge this on Friday if no one has further feedback.

**Pull request merged**